jobspec2/test-fixtures/config-compatibility.hcl

```diff
@@ -0,0 +1,77 @@
+job "example" {
+  group "group" {
+    task "task" {
+      config {
+        ## arbitrary structures to compare HCLv1 and HCLv2
+        ## HCLv2 must parse the same way HCLv1 parser does
+
+
+        # primitive attrs
+        attr_string    = "b"
+        attr_int       = 3
+        attr_large_int = 21474836470
+        attr_float     = 3.2
+
+        # lists attrs
+        attr_list_string = ["a", "b"]
+        attr_list_int    = [1, 2, 4]
+        attr_list_float  = [1.2, 2.3, 4.2]
+        attr_list_hetro  = [1, "a", 3.4, { "k" = "v" }]
+        attr_list_empty  = []
+
+        # map attrs
+        attr_map       = { "KEY" = "VAL" }
```

Comment:
In this PR description you described this as:

> Second, in HCLv1, attributes assigned to map are interpreted as single map slices, so

> ```
> # in HCL
> map_attr = { "K" = "V" }
> 
> # parsed as
> {"map_attr": [{"K": "V"}]}
> ```

Does 

```hcl
map_attr = { 
  "K" = "V"
  "K2" = "V2"
}
```

Parse as:
```json
{
  "map_attr": [
    {"K": "V", "K2": "V2"}
  ]
}`
```
 or 
```json
{
  "map_attr": [
    {"K": "V"}, {"K2": "V2"}
  ]
}
```

(which would be weird but I didn't write the HCL2 parser 😀 ). Maybe we want to put multiple k/v pairs here just so that we're exercising that behavior unambiguously?

Reply:
It'll be the first - I'll add a test

```hcl
{
  "map_attr": [
    {"K": "V", "K2": "V2"}
  ]
}
```